### PR TITLE
Update to put owner and repo in query instead of url

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
@rynowak @anurse

**Comment by Pilchie** [comment]
I didn't understand exactly what you meant by putting it in the query string (web newbie here), so I did this.

File: Pages/Incoming.cshtml.cs [hunk 1]
```diff
@@ -23,11 +23,10 @@ public IncomingModel(swaggerClient client, GitHubClient github)
         public IReadOnlyList<IncomingRepo> IncomingRepositories { get; private set; }
         public RateLimit CurrentRateLimit { get; private set; }
 
-        public async Task OnGet(int channelId, string repo)
+        public async Task OnGet(int channelId, string owner, string repo)
         {
-            // The 'repo' is URL encoded (because it's a URL within a URL). Decode it before calling the API.
-            var decodedRepo = WebUtility.UrlDecode(repo);
-            var latest = await _client.GetLatestAsync(decodedRepo, null, null, channelId, null, null, false, ApiVersion10._20190116);
+            var repoUrl = $"https://github.com/{owner}/{repo}";
```
Comment (rynowak) [inline Pages/Incoming.cshtml.cs:28]: You could do something like this too. 

What I was suggesting is that you construct your URL like this:
```
https://dependencyflow.azurewebsites.net/incoming/3?repo=https%3A%2F%2Fgithub.com%2Faspnet%2FAspNetCore
```

**Comment by analogrelay** [comment]
This seems fine. I'm surprised it wasn't working in the production site. I definitely tested it out and it was working fine locally. Perhaps I was using IIS and since we're deploying to Linux it was running on Kestrel? Or the other way around (since I think IIS/ARR is still used in App Service for Linux on the front-end machines).

This seems like a fine solution, though there **do** exist non-GitHub (AzDO-only) repositories in Maestro. We don't really care about them right now though.

**Comment by rynowak** [comment]
image

Either way, servers all do different things with encoded fslashes in a URL path - for *defense in depth*. There are really 3 solutions to it:
1. Don't pass a URL at all (this PR)
2. Pass the URL in the query string, servers are more lenient about that (remove `repo` as a route parameter and it should *just work*)
3. Use a roundtrippable-catch-all route parameter `@page "/incoming/{channelId}/{**repo}` - this will make the rest of the URL (including slashes)

If you choose option 3 you end up writing code that's really coupled to your choice of server unfortunately. Option 3 would work well on IIS because of the decoding behaviour, and might not work on other servers.

**Comment by analogrelay** [comment]
We should tell Maestro then :) https://maestro-prod.westus2.cloudapp.azure.com/131/https:%2F%2Fgithub.com%2Faspnet%2FAspNetCore/latest/graph

**Comment by analogrelay** [comment]
It seems to be working for them though, I guess they're running on IIS and using a proper route.

I like this change anyway, so 🤷‍♂.

**Comment by Pilchie** [comment]
I also verified it locally with IIS Express on Windows and it worked.  I just backed out the change and tried Kestrel on Windows, and that also worked.  I was going to try Kestrel on WSLv2, but networking doesn't seem to work as it should in my WSLv2 instance - the server runs, but I can't connect to it on localhost :(

I think that's enough due diligence...

**Comment by rynowak** [comment]
> 
> 
> It seems to be working for them though, I guess they're running on IIS and using a proper route.

I think they are using URL rewriting to route this to a razor page that doesn't parse the route at all, and then using angular to do a thing.

https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Maestro.Web/Startup.cs#L415

I was also curious about how this was working.

**Pull request merged**